test/execution/server_test.exs

```diff
@@ -142,5 +142,30 @@ defmodule Wanda.Execution.ServerTest do
 
       assert_receive {:DOWN, ^ref, _, ^pid, :normal}
     end
+
+    test "should go down when the timeout function gets called" do
```

Comment:
here we should test that the facts were partially gathered by sending some of the agents `receive_facts`